app/assets/javascripts/components/buttons/form-button-component.js

```diff
@@ -0,0 +1,47 @@
+ManageIQ.angular.app.component('formButtonComponent', {
+  bindings: {
+    angularForm: '=',
+    modelCopy: '=',
+    model: '=',
+    redirectUrl: '=?',
+    newRecord: '<',
+    entity: '=?',
+    entityName: '=?',
+    saveClicked: '&?',
+    addClicked: '&?',
+    resetClicked: '&?',
+    cancelClicked: '&?',
+  },
```

Comment:
I have now added `resetClicked` and `cancelClicked` to the mix as well.
This gives us complete flexibility of where the methods can be defined.

If it's a standard `Reset`/`Cancel` operation, go with the default.
If it's a non-standard `Reset`/`Cancel` operation, all you need to do is define this in the haml where you call your `form-button-component`, and implement the methods in your main form.
```ruby
"reset-clicked"  => "vm.resetClicked",
"cancel-clicked" => "vm.cancelClicked",
```